wolfcrypt/src/port/Renesas/renesas_tsip_util.c

```diff
@@ -166,6 +187,333 @@ static byte _tls2tsipdef(byte cipher)
+
+
+    if (bench_buf_size % 16)
+        bench_buf_size += 16 - (bench_buf_size % 16);
+    
+    /* buffers passed to TSIP must be 32bit-aligned */  
+    byte* bench_plain  = (byte*)XMALLOC((size_t)bench_buf_size + 16, NULL, DYNAMIC_TYPE_CIPHER);
+    byte* bench_cipher = (byte*)XMALLOC((size_t)bench_buf_size + 16, NULL, DYNAMIC_TYPE_CIPHER);
+    byte* bench_addtn  = (byte*)XMALLOC((size_t)16, NULL, DYNAMIC_TYPE_CIPHER);
+    byte* bench_tag    = (byte*)XMALLOC((size_t)16, NULL, DYNAMIC_TYPE_CIPHER);
+
+    XMEMCPY(bench_addtn,    add_auth_data,AES_AUTH_ADD_SZ );
+    XMEMSET(bench_plain,    0x30, bench_buf_size +16);
+    XMEMSET(bench_cipher,   0,    bench_buf_size +16);  
+    XMEMSET(bench_tag,      0,    AES_AUTH_TAG_SZ);
+    XMEMSET(&aes,           0,    sizeof(aes));
+    
+
+    /* 
+     * The key passed as a parameter of this function should be "installed"
+     * into TSIP with a provisioning key. To avoid this troublesom,   
+     * make TSIP generate AES key in stead and use it for encryption and
+     * decryption.
+     */
+    aes.ctx.keySize = keySz;
+    
+    if(keySz == 16){
+        ret = R_TSIP_GenerateAes128RandomKeyIndex(&aes.ctx.tsip_keyIdx);
+    }
+    else{
+        ret = R_TSIP_GenerateAes256RandomKeyIndex(&aes.ctx.tsip_keyIdx);
+    }
+
+    if( ret!= TSIP_SUCCESS)
+    {
+        WOLFSSL_MSG("R_TSIP_GenerateAesXXXRandomKeyIndex failed" );
+        ret = -1;
+        goto exit;
+    }
+  
+    bench_stats_start(&count, &start);
+
+    do {
+        for (times = 0; times < numBlocks; times++ ) {
+
+            ret = wc_tsip_AesGcmEncrypt(
+                    &aes,            
+                    bench_cipher,
+                    bench_plain,
+                    bench_size,
+                    (byte*)iv,      ivSz, 
+                    bench_tag,      AES_AUTH_TAG_SZ,
+                    bench_addtn,    AES_AUTH_ADD_SZ);
+    
+            if( ret != 0 ){
+                WOLFSSL_MSG("wc_tsip_AesGcmEncrypt failed" );
+                goto exit;
+            }
+        }
+        count += times;
+    } while (bench_stats_sym_check(start));
+   
+    bench_stats_sym_finish(encLabel, doAsync, count, bench_size, start, ret);
+
+
+    bench_stats_start(&count, &start);
+
+    do {
+        for (times = 0; times < numBlocks ; times++) {
+ 
+            ret = wc_tsip_AesGcmDecrypt(
+                    &aes,               
+                    bench_plain,
+                    bench_cipher,
+                    bench_size,
+                    (byte*)iv,      ivSz,
+                    bench_tag,      AES_AUTH_TAG_SZ,
+                    bench_addtn,    AES_AUTH_ADD_SZ);
+            if( ret != 0){
+                WOLFSSL_MSG("wc_tsip_AesGcmDecrypt failed");
+                goto exit;
+            }
+        }
+        count += times;
+    } while (bench_stats_sym_check(start));
+
+    bench_stats_sym_finish(decLabel, doAsync, count, bench_size, start, ret);
+
+
+    (void)decLabel;
+
+exit:
```

Comment:
See note above and coding standards about use of goto

Reply:
Rewrote code without "goto".